File: DSA/N_PARTH_PANDEY_590012761/Day11Question3.py
def count_palindromic_substrings(s: str) -> int:
    count = 0
    n = len(s)
    
    for i in range(n):
        
        left, right = i, i
        while left >= 0 and right < n and s[left] == s[right]:
            count += 1
            left -= 1
            right += 1
        
        
        left, right = i, i + 1
        while left >= 0 and right < n and s[left] == s[right]:
            count += 1
            left -= 1
            right += 1
    
    return count


sample_input = "abc"
sample_output = count_palindromic_substrings(sample_input)
print(f"Input: {sample_input}")
print(f"Output: {sample_output}")

File: DSA/ShatakshiMishra_590015008/Day11Question3.cpp
#include <iostream>
#include <string>

using namespace std;

int countPalindromicSubstrings(string s) {
    int n = s.length();
    int count = 0;

    auto expandAroundCenter = [&](int left, int right) {
        while (left >= 0 && right < n && s[left] == s[right]) {
            count++;
            left--;
            right++;
        }
    };

    for (int i = 0; i < n; ++i) {
        expandAroundCenter(i, i);
        expandAroundCenter(i, i + 1);
    }

    return count;
}

int main() {
    string input;
    cin >> input;

    int result = countPalindromicSubstrings(input);
    cout << result << endl;
}
